src/kibana/components/persisted_state/persisted_state.js

```diff
@@ -0,0 +1,105 @@
+define(function (require) {
+  var _ = require('lodash');
+  var errors = require('errors');
+
+  return function () {
+    function validateParent(parent, path) {
+      if (_.isUndefined(parent)) return;
+
+      if (path.length <= 0) {
```

Comment:
That's what the internal `this._setPath()` is doing - it turns whatever is passed in as the path into an array. If you pass in a number, like `0`, it becomes `[0]` by the time it tries to validate the path.

Either check works, really